### DISCUSSION TITLE
Sample usages for developers

### DISCUSSION THREAD

**Discussion by babakansari** [glideapps/glide-data-grid#251] [Ideas]
Hello,

Could the code for [grid.glideapps.com](https://grid.glideapps.com) be available for developers as a reference? How about sample usages in [Storybook](https://glideapps.github.io/glide-data-grid)?

Is scrolling event from `DataEditorContainer` the preferred way to capture Scroll Top position or there is a better way?

Thanks!

**Comment by jassmith** [comment]
There are quite a few examples in the storybook codebase and storybook itself has a docs section which goes through basic use cases. In 4.0.0 the DataEditorContainer is gone all together, capturing the visible region can be done by responding to the onVisibleRegionChanged callback.

It doesn't give out the scrolltop exactly, and it doesn't do that because in the event of scroll region compression (more content than the browser allows in scrolling regions) the two may not agree. What do you need the scrollTop for?

**Reply by babakansari** [reply]
Thanks for the quick response! 👍 

I need two (or more) grids visible on the page and have scrolling one grid scroll the other grid(s) simultaneously. 

It was a bit challenging to dig into to the code to realize how Context menu can be used in the code. Also, not sure how selected cell modification using keystroke could be implemented but features are available from the provided demos.

Glide grid seems to have superiority over other grids, having sample code under each demo is the only minor missing part IMO.

Cheers!

**Reply by jassmith** [reply]
I will try to see if I can get the same code in place in the storybook. It's not that easy because the dummy data source we use is pretty dumb and I don't think I want people to view that as the "correct" way. It's just a data source that can self-generate data.

As for scroll syncing, let me think about that for a hot minute.

**Reply by jassmith** [reply]
What do you mean "how selected cell modification using keystroke could be implemented"? What are you looking for in particular?

**Reply by babakansari** [reply]
> What do you mean "how selected cell modification using keystroke could be implemented"? What are you looking for in particular?

I want to modify the selected cells value on a key press. Like selecting a cell range and press `D` to assign value `D` to all the selected cells.

**Comment by jassmith** [comment]
Have you looked into this series? https://glideapps.github.io/glide-data-grid/?path=/story/glide-data-grid-docs--getting-started

**Reply by babakansari** [reply]
For the most part it is fine. I couldn't find a hint for context menu implementation, Should it be handled by the mouse click on the cell manually?

**Reply by jassmith** [reply]
I added a section on menus. It could afford to be expanded but it should help you get started.
